app/controllers/data_controller.rb

```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# :nodoc:
+class DataController < ApplicationController
```

Comment:
Instead of creating a new controller for this, I think using the existing `PagesController` would be appropriate because it's a static content page. Just add a new action `def data; end` to the pages controller.

Reply:
Authorization in the controller action could look like:

```
  def data
    authorize! :read, :pages_data
  end
```